WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -626,7 +685,16 @@ def submitJobs( self ):
           return result
 
         executable, pilotSubmissionChunk = result['Value']
-        result = ce.submitJob( executable, '', pilotSubmissionChunk )
+       
+        #check the parameter
+        processors = self.pilotProcessors
+        if self.wholeNode == 'true':
+          processors = -1
+        if processors != 1:
```

Comment:
We still need time to cover all the CE types, so keep the old function to support the single-core-only CEs we haven't covered. It will be removed after all the CE types can support multi-core in DIRAC.